arista-netdevops-community/anta
# feat(anta.cli): Check catalog functionality

# Description

- Add `anta check catalog -c <CATALOG>`
- Add `AntaCatalog` class to manipulate test catalog
   - `tests` property can be used to get and set tests to the catalog.
   - `get_tests_by_tags` to filter tests by tag
   - `get_tests_by_device` to filter tests by device
- Add `AntaCatalogFile` pydantic model to provide validation for test catalog YAML file
    ``` python
    catalog = AntaCatalog(filename='/mnt/lab/my_anta_catalog.yaml')
    catalog.check()
    ```
- Add `AntaTestDefinition` pydantic model to provide validation mainly for programmatic `AntaCatalog` instantiation.
    ``` python
    AntaCatalog(tests=[(AntaTest, AntaTest.Input())])
    ```

Fixes #408
Fixes #407 

# Checklist:



- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] New and existing unit tests pass locally with my changes (`tox -e testenv`)
